code/processes/wdb.q

```diff
@@ -203,27 +203,32 @@ endofdaysave:{[dir;pt]
 	.lg.o[`savefinish;"finished saving data to disk"];
 	};
 
-/- add entries to dictionary of callbacks. if timeout has expired or d now contains all expected rows then it releases each waiting process
+/- add entries to keyed table of callbacks. if timeout has expired or d now contains all expected rows then it releases each waiting process
 handler:{
-	.wdb.d[.z.w]:x;
-	if[(.proc.cp[]>.wdb.timeouttime) or (count[.wdb.d]=.wdb.countreload);
-		.lg.o[`handler;"releasing processes"];
-		.wdb.flushend[];
-		.wdb.d:()!()];
+	 if[not .z.w in  key .wdb.d;
+        .wdb.d[.z.w]:x];
+	if[(.proc.cp[]>.wdb.timeouttime) or (.wdb.countreload=count .wdb.d);
+        	.lg.o[`reload;string[count select from .wdb.d where status=1]," out of ", string[count .wdb.d]," processes successfully reloaded"];
```

Comment:
Would be nice to print the errors of those that failed here for completeness

Reply:
Have added an if statement to print out process which failed to reload and the error. Would it be better to print out the failed processes and errors in the summary at the end - rather than what I have added now which should print out the error after the reload message is sent?